crates/matrix-sdk-crypto/src/types/cross_signing_key.rs

```diff
@@ -68,6 +71,11 @@ impl CrossSigningKey {
     pub fn to_raw<T>(&self) -> Raw<T> {
         Raw::from_json(to_raw_value(&self).expect("Coulnd't serialize cross signing keys"))
     }
+
+    /// Get the first key id and matching Ed25519 key we can find.
+    pub fn get_first_key_and_id(&self) -> Option<(&DeviceKeyId, Ed25519PublicKey)> {
+        self.keys.iter().find_map(|(id, key)| key.ed25519().map(|k| (id.as_ref(), k)))
```

Comment:
Using `?` might be a little easier to read than the nested closure:

```suggestion
        self.keys.iter().find_map(|(id, key)| Some((id.as_ref(), key.ed25519()?)))
```